src/parser/mod.rs

```diff
@@ -11931,10 +11949,41 @@ impl<'a> Parser<'a> {
                 replace_into,
                 priority,
                 insert_alias,
+                settings,
+                format_clause,
             }))
         }
     }
 
+    // Parses format clause used for [ClickHouse]. Formats are different when using `SELECT` and
+    // `INSERT` and also when using the CLI for pipes. It may or may not take an additional
+    // expression after the format so we try to parse the expression but allow failure.
+    //
+    // Since we know we never take an additional expression in `SELECT` context we never only try
+    // to parse if `can_have_expression` is true.
+    //
+    // <https://clickhouse.com/docs/en/interfaces/formats>
```

Comment:
ah so per earlier comment I'm imagining if we use different representations for the format clauses the impl isn't affected?